From f773622bf47caea0ca3a46596d1d607acfbf7134 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 9 Nov 2023 15:46:51 +0100
Subject: [PATCH] arm64: dts: st: OP-TEE async notif on PPI 15 for stm32mp25

Define GIC PPI 15 (aka GIC interrupt line 31) for OP-TEE asynchronous
notification.

Change-Id: Id5216132ae78fb937661be13dd03e5064f7ea04b
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -91,9 +91,13 @@
 	};
 
 	firmware {
-		optee {
+		optee: optee {
 			compatible = "linaro,optee-tz";
 			method = "smc";
+			interrupt-parent = <&intc>;
+			interrupts = <GIC_PPI 15 (GIC_CPU_MASK_SIMPLE(1) | IRQ_TYPE_LEVEL_LOW)>;
+			interrupt-controller;
+			#interrupt-cells = <1>;
 		};
 
 		scmi {
